gating/check/pre_deploy.sh

```diff
@@ -60,10 +60,15 @@ if [[ ! -e phantomjs-2.1.1-linux-x86_64.tar.bz2 ]]; then
   tar -xjf phantomjs-2.1.1-linux-x86_64.tar.bz2
 fi
 
+# FrankZhang: Limit venv python to python2 
+# to avoid python3 code making syntax error https://rpc-openstack.atlassian.net/browse/RI-458
 if [[ ! -d ".venv" ]]; then
-  virtualenv .venv
+  virtualenv .venv --python=python2 --always-copy --no-download
 fi
 
+# FrankZhang: Upgrade pip and setuptools due to outdate issue https://rpc-openstack.atlassian.net/browse/RI-458
+pip install -c{env:UPPER_CONSTRAINTS_FILE:https://git.openstack.org/cgit/openstack/openstack-ansible/tree/global-requirement-pins.txt?h=newton-eol} --upgrade pip setuptools
```

Comment:
This can just be:

```
pip install --upgrade pip setuptools -c https://raw.githubusercontent.com/rcbops/openstack-ansible/stable/newton/global-requirement-pins.txt
```

Reply:
yep, that'd be easiest

Reply:
wow, didn't know we got fork of it, updating soon.